src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1979,9 +1980,15 @@ internal void SetMaxLanguageVersion(ProjectId projectId, string? maxLanguageVers
             }
         }
 
-        internal void RefreshProjectExistsUIContextForLanguage(string language)
+        internal void RefreshProjectExistsUIContextForLanguage(string language, CancellationToken cancellationToken)
+            => _foregroundObject.ThreadingContext.JoinableTaskFactory.Run(async () =>
+                await RefreshProjectExistsUIContextForLanguageAsync(language, cancellationToken).ConfigureAwait(false));
+
+        private async Task RefreshProjectExistsUIContextForLanguageAsync(string language, CancellationToken cancellationToken)
         {
             // We must assert the call is on the foreground as setting UIContext.IsActive would otherwise do a COM RPC.
```

Comment:
Update the comment and remove the AssertIsForeground since we're now actively switching?

Reply:
done.